File: backend-python/app/models/rider.py
"""Rider domain models — Sprint 5.2 (Rider MongoDB integration).

Mirrors the camelCase shapes the rider frontend expects (see
backend/src/mock/server.ts "/api/rider/..." handlers and backend/src/rider/*.ts).
"""

from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel


class RiderTimelineStep(BaseModel):
    id: str
    label: str
    time: str
    done: bool


class RiderOrder(BaseModel):
    id: str
    code: str
    taskType: str
    status: str
    customerName: str
    customerPhone: str
    partnerName: str
    partnerPhone: str
    pickupAddress: str
    deliveryAddress: str
    distanceKm: float
    etaMinutes: int
    estimatedEarning: float
    itemCount: int
    slot: str
    placedAt: str
    paymentMode: str
    timeline: List[RiderTimelineStep] = []


class RiderDocumentStatus(BaseModel):
    id: str
    label: str
    status: str


class RiderProfile(BaseModel):
    riderId: str
    fullName: str
    phone: str
    email: str
    city: str
    rating: float
    totalTrips: int
    joinedOn: str
    vehicleType: str
    vehicleNumber: str
    bankName: str
    accountLast4: str
    ifsc: str
    kycStatus: str
    documents: List[RiderDocumentStatus] = []
    isOnline: bool = False
    onlineMinutes: int = 0


class RiderProfileUpdate(BaseModel):
    fullName: Optional[str] = None
    email: Optional[str] = None
    city: Optional[str] = None
    vehicleType: Optional[str] = None
    vehicleNumber: Optional[str] = None
    bankName: Optional[str] = None
    accountLast4: Optional[str] = None
    ifsc: Optional[str] = None


class RiderSettings(BaseModel):
    isOnline: bool = False
    vehicle: str = ""
    plate: str = ""
    notificationsEnabled: bool = True


class RiderDashboard(BaseModel):
    assigned: int
    active: int
    completedToday: int
    earningsToday: int


class RiderWallet(BaseModel):
    accountId: str
    balance: float
    cashbackBalance: float
    rewardPoints: int
    referralCode: str
    referralEarned: float
    bankLast4: str = ""


class RiderWalletTransaction(BaseModel):
    id: str
    title: str
    date: str
    amount: float
    direction: str
    status: str
    kind: str


class RiderEarnings(BaseModel):
    total: float
    orders: int


class RiderNotification(BaseModel):
    id: str
    accountId: str
    title: str
    body: str
    date: str
    read: bool
    kind: str


class RiderHistoryEntry(BaseModel):
    id: str
    code: str
    customerName: str
    partnerName: str
    date: str
    amount: float
    distanceKm: float
    outcome: str


class RiderAnalyticsDay(BaseModel):
    date: str
    deliveries: int
    earnings: float
    distanceKm: float


class RiderRegistrationPayload(BaseModel):
    payload: dict

File: partner-frontend/src/shared/types/rider.ts
// Rider app domain types. These mirror the shared QuickPress backend contracts
// consumed by customer-frontend and partner-frontend. UI only — no backend.

export type RiderSession = {
  riderId: string;
  phone: string;
  fullName: string;
  isVerified: boolean;
  isOnboarded: boolean;
  isNewRider: boolean;
};

export type VehicleType = "bike" | "scooter" | "ev-scooter" | "cycle" | "mini-truck";

export type RiderRegistrationPayload = {
  photoName: string;
  fullName: string;
  phone: string;
  email: string;
  vehicleType: VehicleType;
  vehicleNumber: string;
  drivingLicense: string;
  aadhaar: string;
  accountHolder: string;
  accountNumber: string;
  ifsc: string;
  serviceCity: string;
};

export type RiderDashboard = {
  riderName: string;
  isOnline: boolean;
  todayDeliveries: number;
  todayEarnings: number;
  pendingPickups: number;
  pendingDeliveries: number;
  completedDeliveries: number;
  rating: number;
  onlineMinutes: number;
};

export type RiderTaskType = "pickup" | "delivery";

export type RiderOrderStatus =
  | "assigned"
  | "accepted"
  | "arriving"
  | "picked"
  | "at-partner"
  | "ready-for-delivery"
  | "delivered"
  | "cancelled"
  | "failed";

export type RiderOrder = {
  id: string;
  code: string;
  taskType: RiderTaskType;
  status: RiderOrderStatus;
  customerName: string;
  customerPhone: string;
  partnerName: string;
  partnerPhone: string;
  pickupAddress: string;
  deliveryAddress: string;
  distanceKm: number;
  etaMinutes: number;
  estimatedEarning: number;
  itemCount: number;
  slot: string;
  placedAt: string;
  paymentMode: "online" | "cod";
  timeline: { id: string; label: string; time: string; done: boolean }[];
};

export type RiderWalletSummary = {
  todayEarnings: number;
  weeklyEarnings: number;
  monthlyEarnings: number;
  availableBalance: number;
  bankLast4: string;
  incentiveProgress: number;
};

export type RiderTransaction = {
  id: string;
  title: string;
  date: string;
  amount: number;
  direction: "credit" | "debit";
  status: "success" | "pending";
  kind: "trip" | "incentive" | "withdrawal" | "penalty" | "tip";
};

export type RiderNotificationKind =
  | "new-order"
  | "pickup-reminder"
  | "delivery-reminder"
  | "payment"
  | "system";

export type RiderNotification = {
  id: string;
  kind: RiderNotificationKind;
  title: string;
  body: string;
  time: string;
  unread: boolean;
};

export type RiderProfile = {
  riderId: string;
  fullName: string;
  phone: string;
  email: string;
  city: string;
  rating: number;
  totalTrips: number;
  joinedOn: string;
  vehicleType: VehicleType;
  vehicleNumber: string;
  bankName: string;
  accountLast4: string;
  ifsc: string;
  kycStatus: "verified" | "pending" | "rejected";
  documents: { id: string; label: string; status: "verified" | "pending" | "rejected" }[];
};

export type RiderHistoryEntry = {
  id: string;
  code: string;
  customerName: string;
  partnerName: string;
  date: string;
  amount: number;
  distanceKm: number;
  outcome: "completed" | "cancelled" | "failed";
};
